### DISCUSSION TITLE
View presets

### DISCUSSION THREAD

**Discussion by curtistornade** [demberto/PyFLP#125] [General]
Hi !
I was wondering if it would be possible to see the preset name of a VST plugin with PyFLP ?

**Comment by demberto** [comment]
VST plugins store their preset data by themselves. So only the plugin itself understands it.

However, many VST plugins still use the old FXP format which does store the preset name in a standardised way:

1. You can get the preset data of a VST plugin with either [`Instrument.plugin.state`](https://pyflp.readthedocs.io/en/latest/reference/channel/instrument.html#pyflp.channel.Instrument.plugin) or [`Slot.plugn.state`](https://pyflp.readthedocs.io/en/latest/reference/mixer/slot.html#pyflp.mixer.Slot.plugin) *both of which are actually [`VSTPlugin.state`](https://pyflp.readthedocs.io/en/latest/reference/plugins/vst.html#pyflp.plugin.VSTPlugin.state)*.
2. Use [this `construct`](https://github.com/demberto/fxp/blob/d43a0468ad37ecb7b75ca0b697e02748b9922df5/fxp/__init__.py#L68-L86) definition to parse the state *or* use my library [fxp](https://pypi.org/project/fxp/) to parse the plugin preset file.

If it fails to parse you are out of luck and you need to reverse-engineer the plugin preset format. It might be JSON, XML or some propreitary binary format. Some plugins don't even save the preset name inside their project file / preset data.

**Comment by curtistornade** [comment]
Okay thank you very much
